### ISSUE TITLE
Salesforce BigQuery Transfer

### ISSUE THREAD

**Issue by RomeoAva** [terraform-google-modules/terraform-google-bigquery#329]
### TL;DR

Hello, I am trying to create a [Salesforce Transfer](https://cloud.google.com/bigquery/docs/salesforce-transfer) for BQ using Terraform. Here is my current config to create that ressource:

```
resource "google_bigquery_data_transfer_config" "query_config" {
  display_name           = "import_salesforce"
  location               = "US"
  data_source_id         = "salesforce"
  destination_dataset_id = "analysis"
  params = {
    "connector.authentication.oauth.clientId"     = ""
    "connector.authentication.oauth.clientSecret" = ""
    "connector.authentication.username"           = ""
    "connector.authentication.password"           = ""
    "connector.authentication.securityToken"      = ""
    "assets"                                      = ["Project_ABC"]
  }
}
```

### Expected behavior

To create a Salesforce transfer pulling the requested assets.

### Observed behavior

When specifying a list in the `assets` parameter, terraform tells me:

```
 Inappropriate value for attribute "params": element "assets": string required.
```

When I try to convert it to a String, the transfer is created but on BigQuery I get the following error:

```
Incorrect assets set in transfer config : string_value: "Project_ABC" , assets should be a non-empty list of strings.
```

I also tried to json_encode the list, but same result on BQ side:

```
Incorrect assets set in transfer config : string_value: "[\"Project_ABC\"]" , assets should be a non-empty list of strings.
```

### Terraform Configuration

```hcl
terraform {
  backend "gcs" {
    bucket = ""
    prefix = "terraform/state"
  }

  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "5.29.1"
    }
    random = {
      source = "hashicorp/random"
    }
  }
}
```

### Terraform Version

```sh
Terraform v1.5.7
```

**Comment by RomeoAva** [comment]
Sorry I think this is the wrong repo
